test/distributed/_tensor/test_math_ops.py

```diff
@@ -112,22 +108,19 @@ def test_softmax_with_bwd(self):
 
             dist_x = distribute_tensor(x, device_mesh, [Shard(shard_dim)])
             self.assertTrue(dist_x.requires_grad)
+            dist_softmax = dist_x.softmax(dim=softmax_dim)
+            shard_dim = shard_dim + dist_x.ndim if shard_dim < 0 else shard_dim
+            dist_y = dist_softmax.sum()
             if dims[softmax_dim] == dims[shard_dim]:
-                with self.assertRaisesRegex(
-                    Exception, "Cannot run .* on sharding dimension!$"
-                ):
-                    dist_softmax = dist_x.softmax(dim=softmax_dim)
+                self.assertTrue(dist_y.placements[0].is_replicate())
             else:
-                dist_softmax = dist_x.softmax(dim=softmax_dim)
-                shard_dim = shard_dim + dist_x.ndim if shard_dim < 0 else shard_dim
                 self.assertTrue(dist_softmax.placements[0].is_shard(dim=shard_dim))
-                dist_y = dist_softmax.sum()
                 dist_y = dist_y.redistribute(device_mesh, [Replicate()])
-                self.assertEqual(dist_y.to_local(), local_y)
-                self.assertIsNone(dist_x.grad)
-                dist_y.backward()
-                self.assertIsNotNone(dist_x.grad)
-                self.assertEqual(dist_x.grad.full_tensor(), x.grad)
+            self.assertEqual(dist_y.to_local(), local_y)
```

Comment:
I suggest we also check `dist_y.grad`'s sharding placements as well before redistributing.

Reply:
Hmm there seems no `.grad` at this place. did you mean `dist_y`'s sharding placements? If so I agree!